auth/auth-server/src/telemetry/helpers.rs

```diff
@@ -0,0 +1,154 @@
+}
+
+/// Record a volume metric with the given extra tags
+fn record_volume_with_tags(
+    mint: &str,
+    amount: u128,
+    volume_metric_name: &'static str,
+    extra_labels: &[(String, String)],
+) {
+    let (asset, volume) = get_asset_and_volume(mint, amount);
+    let mut labels = vec![(ASSET_METRIC_TAG.to_string(), asset)];
+    let extra_labels = extra_labels.iter().map(|(k, v)| (k.clone(), v.clone()));
+    labels.extend(extra_labels);
+
+    // We use a gauge metric here to be able to capture a float value
+    // for the volume
+    metrics::gauge!(volume_metric_name, labels.as_slice()).set(volume);
+}
+
+/// Records metrics for the incoming external match request
+fn record_external_match_request_metrics(
+    req: &ExternalMatchRequest,
+    price: f64,
+    labels: &[(String, String)],
+) -> Result<(), AuthServerError> {
+    // Record external order volume
+    let base_mint = biguint_to_hex_addr(&req.external_order.base_mint);
+    let quote_mint = biguint_to_hex_addr(&req.external_order.quote_mint);
+
+    // Calculate amount in base
+    let timestamped_price = TimestampedPrice::new(price);
+    let fixed_point_price = timestamped_price.as_fixed_point();
+    let order = req.external_order.to_order_with_price(fixed_point_price);
+
+    // Calculate amount in quote
+    let (_, volume) = get_asset_and_volume(&base_mint, order.amount);
+    let quote_token = Token::from_addr(&quote_mint);
+    let quote_amount = convert_from_decimal(&quote_token, volume * fixed_point_price.to_f64());
+
+    record_volume_with_tags(&base_mint, order.amount, EXTERNAL_ORDER_BASE_VOLUME, labels);
+    record_volume_with_tags(&quote_mint, quote_amount, EXTERNAL_ORDER_QUOTE_VOLUME, labels);
+
+    Ok(())
+}
+
+/// Records metrics for the external match response (match bundle)
+fn record_external_match_response_metrics(
+    resp: &ExternalMatchResponse,
+    labels: &[(String, String)],
+) -> Result<(), AuthServerError> {
+    let (quote, base) = match resp.match_bundle.match_result.direction {
+        OrderSide::Buy => (&resp.match_bundle.send, &resp.match_bundle.receive),
+        OrderSide::Sell => (&resp.match_bundle.receive, &resp.match_bundle.send),
+    };
+
+    record_volume_with_tags(&quote.mint, quote.amount, EXTERNAL_MATCH_QUOTE_VOLUME, labels);
+    record_volume_with_tags(&base.mint, base.amount, EXTERNAL_MATCH_BASE_VOLUME, labels);
+
+    Ok(())
+}
+
+/// Records a counter metric for a given base mint and key description
+fn record_endpoint_metrics(base_mint: &str, key_description: String, metric_name: &'static str) {
+    let (asset, _) = get_asset_and_volume(base_mint, 0);
+    let labels = vec![
+        (ASSET_METRIC_TAG.to_string(), asset),
+        (KEY_DESCRIPTION_METRIC_TAG.to_string(), key_description),
+    ];
+
+    metrics::counter!(metric_name, &labels).increment(1);
+}
+
+/// Records all metrics related to an external match request and response
+pub async fn record_external_match_metrics(
```

Comment:
don't see any `await`s in this method, so it doesn't need to be async. Will you add `#![deny(clippy::unused_async)]` to the top of `main.rs` so that clippy will catch this in the future?